docs/_posts/examples/3400-01-14-filter-features-within-map-view.html

```diff
@@ -0,0 +1,229 @@
+---
+layout: example
+category: example
+title: "Filter features within map view"
+description: "Move the map to query viewable features in a vector tile layer and filter by typing in an input"
+---
+<style>
+.map-overlay {
+    position: absolute;
+    width: 25%;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    padding: 10px;
+}
+
+.map-overlay-inner {
+    font: 12px/20px 'Helvetica Neue', Arial, Helvetica, sans-serif;
+    background-color: #fff;
+    box-shadow: 0 1px 2px rgba(0,0,0,0.10);
+    border-radius: 3px;
+    max-height: 100%;
+    overflow: hidden;
+}
+
+.map-overlay-inner fieldset {
+    display: none;
+    background: #ddd;
+    border: none;
+    padding: 10px;
+    margin: 0;
+}
+
+.map-overlay-inner input {
+    display: block;
+    border: none;
+    width: 100%;
+    border-radius: 3px;
+    padding: 10px;
+    margin: 0;
+}
+
+.map-overlay-inner .listing {
+    overflow: auto;
+    max-height: 100%;
+}
+
+.map-overlay-inner .listing > * {
+    display: block;
+    padding: 5px 10px;
+    margin: 0;
+}
+
+.map-overlay-inner .listing a {
+    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
+    color: #404;
+    text-decoration: none;
+}
+
+.map-overlay-inner .listing a:last-child {
+    border: none;
+}
+
+.map-overlay-inner .listing a:hover {
+    background: #f0f0f0;
+}
+</style>
+
+<div id='map'></div>
+
+<div class='map-overlay'>
+    <div class='map-overlay-inner'>
+        <fieldset>
+            <input id='feature-filter' type='text' placeholder='Filter results by name' />
+        </fieldset>
+        <div id='feature-listing' class='listing'></div>
+    </div>
+</div>
+
+<script>
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/streets-v8',
+    center: [-98, 38.88],
+    maxZoom: 5,
+    minZoom: 1,
+    zoom: 3
+});
+
+// Holds visible airport features for filtering
+var airports = [];
+
+// Create a popup, but don't add it to the map yet.
+var popup = new mapboxgl.Popup({
+    closeButton: false
+});
+
+var filter = document.getElementById('feature-filter');
+var listing = document.getElementById('feature-listing');
+
+function renderFeatures(features) {
```

Comment:
What do you think about renaming these identifiers to differentiate things that have to do with Mapbox domain objects from things that have to do with HTML domain objects?

 - `filter` -> `filterView`
 - `listing` -> `featuresView`
 - `renderFeatures` -> `renderFeaturesView`